File: 6KYU/Tank Truck/solution.clj
(ns voltank.core)

(defn tank-vol [h d vt]
  (let [r (/ d 2.0)
        r-sq (* r r)
        l (/ vt (* Math/PI r-sq))]
    (int
     (Math/floor
      (* l (- (* r-sq
                 (Math/acos (/ (- r h)
                               r)))
              (* (- r h)
                 (Math/sqrt (- (* 2.0 r h)
                               (* h h))))))))))

File: 6KYU/Tank Truck/solution.cs
using System;

public class VolTank
{
  public static int TankVol(int h, int d, int vt)
  {
       var r = d/2.0;
       var l = vt / (Math.PI * (r * r));
       return (int)Math.Floor(l * ((r * r) * Math.Acos((r - h) / r) - ((r - h) * Math.Sqrt((2 * r * h) - (h * h)))));
   }
}